examples/gno.land/p/samourai/daocond/daocond.gno

```diff
@@ -0,0 +1,29 @@
+package daocond
+
+// We made conditions stateless to avoid complexity of handling events like (member added, member removed, role assigned, role removed, etc.)
+// This model should work pretty good for small DAOs with few voters.
+// In future, we should probably add a stateful model for large DAOs with many voters.
+
+type Condition interface {
+	Eval(ballot Ballot) bool
+	Signal(ballot Ballot) float64
+
+	Render() string
+	RenderWithVotes(ballot Ballot) string
+}
+
+type Ballot interface {
+	Vote(voter string, vote Vote)
+	Get(voter string) Vote
+	Total() int
+
+	Iterate(fn func(voter string, vote Vote))
```

Comment:
```suggestion
	Iterate(fn func(voter string, vote Vote)) bool
```
the bool means "stop" or "continue" like avl, so we don't have to iterate through the whole ballot if we're looking for a subset of elements (for example one member)

Reply:
also could be nice to have
```suggestion
	IterateKind(voteKind Vote, fn func(voter string, vote Vote) bool)
```
because I think in all current condition implems we only care about "yes" votes
but maybe this is an ealry optimization and the cost of maintaining the indexes for vote kind might be more costly